src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -299,14 +301,24 @@ public function countMessagesInQueues(): int
      */
     private function publishWithDelay(string $body, array $headers, int $delay, AmqpStamp $amqpStamp = null)
     {
+        if ($this->shouldSetup()) {
+            $this->setup(); // setup delay exchange and normal exchange for delay queue to DLX messages to
+        }
+
         $routingKey = $this->getRoutingKeyForMessage($amqpStamp);
+        $originQueue = $headers['X-Origin-Queue'] ?? null;
```

Comment:
the header is only needed because we don't have access to the AmqpReceivedStamp here. but we do not want to actually send it or? so this should be remove again.

Reply:
i think it should not be removed again, because dead letter queues names or routing keys not always tell us from what queue message comes from. So it can be used as the debug information.
But i am fine with passing this information in some other way :)